src/cli.rs

```diff
@@ -456,3 +522,57 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct NodeConfig {
+    pub data_dir: String,
+    pub pubkey: String,
+    pub macaroon: String,
+    pub role: u8,
+    pub token: String,
+}
+
+impl Default for NodeConfig {
+    fn default() -> Self {
+        
+        NodeConfig {
+            data_dir: ".".into(),
+            pubkey: "".into(),
+            macaroon: "".into(),
+            role: Role::Admin.to_integer(),
+            token: "satoshi".into(),
+        }
+    }
+}
+
+impl NodeConfig {
+    pub fn path(&self) -> String {
+        dbg!("Looking up a path");
+        format!("{}/data/{}", self.data_dir, self.pubkey)
+    }
+
+    pub fn save(&mut self) {
+        dbg!("Trying to save");
+        fs::write(
+            self.path().clone(),
+            serde_json::to_string(&self).expect("failed to serialize config"),
+        )
+        .expect("failed to write config");
+        dbg!("Saved");
+    }
+}
+
+// I could not figure out how to improt this from the database/admin
```

Comment:
It may be easy to import this Role from the database::admin but for whatever reason I just couldn't make it happen.